File: docs/examples/1_login.ts
import axios from 'axios';

const apiClient = axios.create({
  baseURL: 'http://localhost:3000',
  headers: {
    'Content-Type': 'application/json',
  },
});

async function login() {
  try {
    const response = await apiClient.post('/auth/login', {"email":"learner@example.com","password":"Password123!"});
    
    console.log('Response:', response.data);
    return response.data;
  } catch (error) {
    console.error('Error:', error.response?.data || error.message);
  }
}

// Usage
login();

File: docs/examples/1_login.py
import requests
import json

def login():
    url = "http://localhost:3000/auth/login"
    
    data = {
        "email": "learner@example.com",
        "password": "Password123!"
    }
    
    try:
        response = requests.post(url, json=data)
        response.raise_for_status()
        print("Response:", response.json())
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Error: {e}")
        return None

# Usage
if __name__ == "__main__":
    login()